File: ce/snippets/visualbasic/CRMV8/solutionsvb/vb/workwithsolutions5.vb
' <snippetworkwithsolutions5>


 ' Add an existing Solution Component
 'Add the Account entity to the solution
 Dim retrieveForAddAccountRequest As New RetrieveEntityRequest() With {
  .LogicalName = Account.EntityLogicalName
 }
 Dim retrieveForAddAccountResponse As RetrieveEntityResponse =
  CType(_serviceProxy.Execute(retrieveForAddAccountRequest), 
   RetrieveEntityResponse)
 Dim addReq As New AddSolutionComponentRequest() With {
  .ComponentType = componenttype.Entity,
  .ComponentId = CType(retrieveForAddAccountResponse.EntityMetadata.MetadataId, Guid),
  .SolutionUniqueName = solution.UniqueName
 }
 _serviceProxy.Execute(addReq)

' </snippetworkwithsolutions5>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/solutions/cs/workwithsolutions5.cs
// <snippetworkwithsolutions5>


 // Add an existing Solution Component
 //Add the Account entity to the solution
 RetrieveEntityRequest retrieveForAddAccountRequest = new RetrieveEntityRequest()
 {
     LogicalName = Account.EntityLogicalName
 };
 RetrieveEntityResponse retrieveForAddAccountResponse = (RetrieveEntityResponse)_serviceProxy.Execute(retrieveForAddAccountRequest);
 AddSolutionComponentRequest addReq = new AddSolutionComponentRequest()
 {
     ComponentType = (int)componenttype.Entity,
     ComponentId = (Guid)retrieveForAddAccountResponse.EntityMetadata.MetadataId,
     SolutionUniqueName = solution.UniqueName
 };
 _serviceProxy.Execute(addReq);

// </snippetworkwithsolutions5>